packages/grid/x-data-grid/src/components/panel/filterPanel/GridFilterForm.tsx

```diff
@@ -286,7 +287,9 @@ function GridFilterForm(props: GridFilterFormProps) {
           onClick={handleDeleteFilter}
           size="small"
         >
+          {rootProps.components.FilterPanelDeleteIcon?.name ? 
           <rootProps.components.FilterPanelDeleteIcon />
+          : <GridCloseIcon fontSize="small" />}
```

Comment:
Why using `rootProps.components.FilterPanelDeleteIcon?.name` as a criterion? Any icon created by `createSvgIcon()` will have a name.

A simpler solution would be to set `<rootProps.components.FilterPanelDeleteIcon fontSize="small" />`

The drawback is that all the icons will receive the prop `fontSize="small"` with no way to change it.

If developers want to use big icons, they can ignore the prop as follow:
```jsx
const MyBigIcon = () => <CloseIcon fontSize="large" />

<DataGrid components={{ FilterPanelDeleteIcon: MyBigIcon }} />
```

@Hameezr  Does that sound good to you?

Reply:
Yes, this will solve the problem and add some customizability as well! Thanks for the suggestion, will make this change